src/compiler/emitter.ts

```diff
@@ -3052,6 +3054,17 @@ module ts {
                     write(": ");
                     // Even though this is stored as identifier treat it as an expression
                     // Short-hand, { x }, is equivalent of normal form { x: x }
+
+                    if (languageVersion < ScriptTarget.ES6) {
+                    // short-hand property can be initialized with renamed let binding
+                    // in this case use proper name
+                        var generatedName = getGeneratedNameForIdentifier(node.name, /*isValueOfShorthandPropertyAssignment*/ true);

```

Comment:
I might suggest a different approach. Instead of passing a boolean to the resolver to say whether you want a rename, the resolver should always give you the block scope variable id. Then in emitShorthandPropertyAssignment, you can decide whether or not to use it. In other words, when you emit the value here, you can call emitExpressionIdentifier, and make sure it calls into the resolver to get a generated name. But when you emit the property name, instead of saying `emit(node.name)`, you can just write the source text of the node without calling emitIdentifier.